README.md

```diff
@@ -10,13 +10,21 @@ non-public information about using this repository.
 
 ### Terraform
 
-The main terraform directories are:
+Our Terraform code is organized by two concepts, with two corresponding directories.
 
-* `modules`: where we decompose our configuration into [Terraform
-  modules](https://www.terraform.io/docs/configuration-0-11/modules.html)
-* `modules/stack/base` & `modules/stack/spoke`: the main modules that define
-  the bulk of each environment
-* `stacks`: the various "environments"
+* **Modules** are reusable units of terraform code. Each module describes a single useful concept in our infrastructure. Modules are parameterized by terraform variables for reuse, and are located in `terraform/modules`.
+  * Two important modules are `modules/stack/base` and `modules/stack/spoke`.
+  * Read more about Terraform modules: https://developer.hashicorp.com/terraform/language/modules
+* **Stacks** combine and configure modules for use in an environment. They are also parameterized by variables, such as the name of the environment. Stacks are located in `terraform/stacks`.
+
+As an example, how could we write terraform code to deploy CloudFront distributions in front of three load balancers in an environment?
+
+* A good `cloudfront` module might declare a CloudFront distribution, a Shield Advanced resource to protect it, and an ACL association between the distribution and an ACL. (The ACL itself is not declared in the module.) It could take an origin, a list of domains, and an ACL ARN as variables.
+* A good `cloudfront` stack might reuse the `cloudfront` module three times, once for each load balancer in the environment. It would pass an external domain and load balancer domain to each module. It could also declare a single ACL for the environment and pass its ARN to each `cloudfront` module. The stack could take an environment name as a variable.
+
+Lastly, you would add a job to Concourse for each environment. Each job would deploy the `cloudfront` stack, passing in the environment name as a variable.
```

Comment:
As an example, if we wanted to write terraform code to deploy several CloudFront distributions in front of three load balancers in an environment, we could:

Create a 'cloudfront' module that declares a CloudFront distribution, a Shield Advanced resource to protect the distribution, and an Access Control List (ACL) association...

Then, create a 'cloudfront' stack that references the 'cloudfront' module and reuses it three times, once for each load balancer...

Lastly, we would add a job to Concourse for each environment, where each job deploys the 'cloudfront' stack and passes in the environment name as a variable.